ql/src/test/org/apache/hadoop/hive/ql/lockmgr/TestDbTxnManager2.java

```diff
@@ -4696,4 +4697,37 @@ public void testMaterializedViewEnableRewriteNonBlocking() throws Exception {
     txnMgr.rollbackTxn();
     driver.run("drop materialized view mv_tab_acid");
   }
+
+  @Test
+  public void testMaterializedViewRebuildLockForSameMV() throws Exception {
+    driver.run("drop materialized view if exists mv_tab_acid");
+    dropTable(new String[]{"tab_acid"});
+
+    driver.run("create table if not exists tab_acid (a int, b int) partitioned by (p string) " +
+        "stored as orc TBLPROPERTIES ('transactional'='true')");
+    driver.run("insert into tab_acid partition(p) (a,b,p) values(1,2,'foo'),(3,4,'bar')");
+
+    driver.run("create materialized view mv_tab_acid partitioned on (p) " +
+        "stored as orc TBLPROPERTIES ('transactional'='true') as select a, p from tab_acid where b > 1");
+
+    driver.run("insert into tab_acid partition(p) (a,b,p) values(1,2,'foo'),(3,4,'bar')");
+
+    driver.compileAndRespond("alter materialized view mv_tab_acid rebuild");
+    driver.lockAndRespond();
+
+    // rollback the transaction
+    txnMgr.rollbackTxn();
+
+    try {
+      driver.compileAndRespond("alter materialized view mv_tab_acid rebuild");
+    }
+    catch (Exception se) {
+      Assert.assertEquals(
+          "FAILED: SemanticException org.apache.hadoop.hive.ql.parse.SemanticException: Another process is rebuilding the materialized view default.mv_tab_acid",
+          se.getMessage());
+      Assert.fail("Should not throw exception here");
+    }
```

Comment:
nit.: `Assert.assertThrows` can be used instead of this.

Reply:
Here I am not expecting exception, thats why Assert.fail() is added in catch block to fail the test case. As per my understanding Assert.assertThrows() is getting used when we expect some exception.